### PR TITLE
Update OTLP Exporter insecure channel docs

### PR DESCRIPTION
**Pull request by pellared**
Clarify the docs.

### CODE REVIEW

File: src/OpenTelemetry.Exporter.OpenTelemetryProtocol/README.md [hunk 1]
```diff
@@ -53,10 +53,8 @@ values of the `OtlpExporterOptions`
 
 ## Special case when using insecure channel
 
-If your application is [.NET Standard
-2.1](https://docs.microsoft.com/dotnet/standard/net-standard) or above, and you
-are using an insecure (http) endpoint, the following switch must be set before
-adding `OtlpExporter`.
+If your application is targeting .NET Core 3.1, and you are using an insecure
```
Comment (reyang) [inline src/OpenTelemetry.Exporter.OpenTelemetryProtocol/README.md:56]: Just to confirm, only ".NET Core 3.1" not ".NET Core 3.1 or above"?

Comment (reyang) [reply src/OpenTelemetry.Exporter.OpenTelemetryProtocol/README.md:56]: Seems that [this doc](https://docs.microsoft.com/en-us/aspnet/core/grpc/troubleshoot?view=aspnetcore-5.0#call-insecure-grpc-services-with-net-core-client) has confirmed that only 3.x is affected。

**Review by reyang — APPROVED** [review]

File: src/OpenTelemetry.Exporter.OpenTelemetryProtocol/README.md (hunk 1 above)
Comment (pellared) [reply src/OpenTelemetry.Exporter.OpenTelemetryProtocol/README.md:56]: Correct and I tested it as well

Comment (reyang) [reply src/OpenTelemetry.Exporter.OpenTelemetryProtocol/README.md:56]: @pellared thanks for confirmation!

**Review by cijothomas — APPROVED** [review]

**Pull request merged**